modules/RouteNode.ts

```diff
@@ -110,44 +110,43 @@ export default class RouteNode {
             return
         }
 
-        if (route instanceof Array) {
-            route.forEach(r => this.add(r, cb))
-            return
-        }
-
-        if (!(route instanceof RouteNode) && !(route instanceof Object)) {
-            throw new Error(
-                'RouteNode.add() expects routes to be an Object or an instance of RouteNode.'
-            )
-        } else if (route instanceof RouteNode) {
-            route.setParent(this)
-            this.addRouteNode(route)
-        } else {
-            if (!route.name || !route.path) {
+        const routes = route instanceof Array ? route : [route]
+        routes.forEach((route, idx) => {
```

Comment:
a `for(;;)` loop will be slightly faster

Reply:
I think that performance win is minimal (unmeasureable in my test). But I can do that if you'd like.

```js
console.time(); 
var data = [...Array(100000)]; 
for (var i = 0; i < data.length; i++) {
  var item = data[i]
}; 
console.timeEnd();
```

=> default: 11.075927734375ms

```js
console.time();
[...Array(100000)].forEach((x, idx) => {}); 
console.timeEnd();
```

=> default: 11.264990234375ms

I don't think it will make any difference in practice and for loops are more prone to errors. Let me know!

Reply:
Yeah the difference is quite minimal. I guess this PR is more than enough for you!

Reply:
Are you able to merge this @ematipico ?